### PR TITLE
nvidia_x11: don't leak the kernel reference into nvidia-settings

### PR DESCRIPTION
**Pull request by K900**
###### Description of changes

This removes the LTS kernel from the build closure of things that just want libxnvctrl, like MangoHud.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by K900** [comment]
Before:

```
nixpkgs master  
❯ nix why-depends --derivation '.#mangohud' '.#linux_5_15'
/nix/store/xib7lqqxbiazipwg63gaa6pqv7md2ik7-mangohud-0.6.8.drv
└───/nix/store/nm2yv5y560jyasi5kgxkhdr61vlii9w6-libXNVCtrl-525.85.05.drv
    └───/nix/store/5nxvgwr99n2v1aw0dpvsbdq8nn8big3v-linux-5.15.91.drv
```

After:

```
nixpkgs nvidia-kernel-deps-leak
❯ nix why-depends --derivation '.#mangohud' '.#linux_5_15'
'git+file:///home/k900/nixpkgs#mangohud' does not depend on 'git+file:///home/k900/nixpkgs#linux_5_15'
```

**Comment by teto** [comment]
just curious, what's the impact  in terms of closure size ?

**Comment by K900** [comment]
It's build time, so no impact on installed size.

File: pkgs/os-specific/linux/nvidia-x11/settings.nix [hunk 1]
```diff
@@ -25,7 +25,7 @@ let
       cd src/libXNVCtrl
     '';
 
-    makeFlags = nvidia_x11.makeFlags ++ [
+    makeFlags = nvidia_x11.kernel.makeFlags ++ [
```
Comment (NickCao) [inline pkgs/os-specific/linux/nvidia-x11/settings.nix:28]: Are these kernel.makeFlags even required, I thought they are only useful for kernel modules.

Comment (K900) [reply pkgs/os-specific/linux/nvidia-x11/settings.nix:28]: The original change that introduced these says these need to match the compiler the kernel is built with: https://github.com/NixOS/nixpkgs/pull/136247

**Comment by K900** [comment]
So, update on this: I've tried to test this using @lovesegfault's LTO kernel overlay, and it doesn't build, but neither does master, so I'm just going to merge this and if anyone wants to fix it to build with LTO, we can always do it later.

**Pull request merged**